src/Umbraco.Core/Services/Implement/FileService.cs

```diff
@@ -539,6 +554,12 @@ public IEnumerable<ITemplate> GetTemplateChildren(int masterTemplateId)
         /// <param name="userId"></param>
         public void SaveTemplate(ITemplate template, int userId = Constants.Security.SuperUserId)
         {
+            if (template.Name != null && template.Name.Length > 255)
```

Comment:
Let's also change this to:
```suggestion
            if (template == null)
            {
                throw new ArgumentNullException(nameof(template));
            }
            
            if (string.IsNullOrWhiteSpace(template.Name) || template.Name.Length > 255)
```